File: 411final/411final/src/main/java/com/example/demo/entity/StockPrice.java
package com.example.demo.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
public class StockPrice {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long Id;
    public String ticker;
    @Column(name="tradeDate")
    public Date date;
    public Double volume;
    public Double low;
    public Double high;
    public Double open;
    public Double close;
}

File: src/app/shared/models/stock-price.ts
export class StockPrice {
  id: number;
  ticker: string;
  volume: number;
  low: number;
  high: number;
  open: number;
  close: number;
  date: Date;

  constructor(data: any) {
    Object.assign(this, data);

    // Assigning date
    if (data.date) {
      this.date = new Date(data.date);
    }
  }
}
